mdoc/src/main/scala-3/mdoc/internal/markdown/SectionInput.scala

```diff
@@ -0,0 +1,68 @@
+package mdoc.internal.markdown
+
+import scala.meta.inputs.Position
+
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.parsing.Parsers.Parser
+import dotty.tools.dotc.util.SourceFile
+import dotty.tools.dotc.util.ScriptSourceFile
+
+import mdoc.internal.cli.{Context => MContext}
+import dotty.tools.dotc.ast.untpd._
+import scala.meta.Source
+import scala.meta.inputs.Input
+import scala.meta.Name
+import mdoc.internal.pos.TokenEditDistance
+
+import dotty.tools.dotc.interactive.InteractiveDriver
+case class SectionInput(input : Input, mod : Modifier, context : MContext){
```

Comment:
The code is currently parsed here and we extract the statements by wrapping the worksheet code inside an object. Needed only for parsing.